tests/e2e-tests/specs/front-end/front-end-checkout.test.js

```diff
@@ -0,0 +1,151 @@
+
+		await expect( page ).toFill( '#billing_first_name', 'John' );
+		await expect( page ).toFill( '#billing_last_name', 'Doe' );
+		await expect( page ).toFill( '#billing_company', 'Automattic' );
+		await expect( page ).toFill( '#billing_email', 'john.doe@example.com' );
+		await expect( page ).toFill( '#billing_phone', '123456789' );
+		await expect( page ).toSelect( '#billing_country', 'United States (US)' );
+		await expect( page ).toFill( '#billing_address_1', 'addr 1' );
+		await expect( page ).toFill( '#billing_address_2', 'addr 2' );
+		await expect( page ).toFill( '#billing_city', 'San Francisco' );
+		await expect( page ).toSelect( '#billing_state', 'California' );
+		await expect( page ).toFill( '#billing_postcode', '94107' );
+	} );
+
+	it( 'allows customer to fill shipping details', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+
+		await expect( page ).toClick( '#ship-to-different-address-checkbox' );
+		await uiUnblocked();
+
+		await expect( page ).toFill( '#shipping_first_name', 'John' );
+		await expect( page ).toFill( '#shipping_last_name', 'Doe' );
+		await expect( page ).toFill( '#shipping_company', 'Automattic' );
+		await expect( page ).toSelect( '#shipping_country', 'United States (US)' );
+		await expect( page ).toFill( '#shipping_address_1', 'addr 1' );
+		await expect( page ).toFill( '#shipping_address_2', 'addr 2' );
+		await expect( page ).toFill( '#shipping_city', 'San Francisco' );
+		await expect( page ).toSelect( '#shipping_state', 'California' );
+		await expect( page ).toFill( '#shipping_postcode', '94107' );
+	} );
+
+	it( 'allows guest customer to place order', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+
+		await expect( page ).toFill( '#billing_first_name', 'John' );
+		await expect( page ).toFill( '#billing_last_name', 'Doe' );
+		await expect( page ).toFill( '#billing_company', 'Automattic' );
+		await expect( page ).toFill( '#billing_email', 'john.doe@example.com' );
+		await expect( page ).toFill( '#billing_phone', '123456789' );
+		await expect( page ).toSelect( '#billing_country', 'United States (US)' );
+		await expect( page ).toFill( '#billing_address_1', 'addr 1' );
+		await expect( page ).toFill( '#billing_address_2', 'addr 2' );
+		await expect( page ).toFill( '#billing_city', 'San Francisco' );
+		await expect( page ).toSelect( '#billing_state', 'California' );
+		await expect( page ).toFill( '#billing_postcode', '94107' );
+		await uiUnblocked();
+
+		await expect( page ).toClick( '.wc_payment_method label', { text: 'Cash on delivery' } );
+		await expect( page ).toMatchElement( '.payment_method_cod', { text: 'Pay with cash upon delivery.' } );
+		await uiUnblocked();
+		await CustomerFlow.placeOrder();
+
+		await expect( page ).toMatch( 'Order received' );
```

Comment:
I wonder if besides just checking if the message is displayed, we should also check if the order actually exists and if its contents match what is expected? This way we could catch more issues then if we simply check for the message.

Reply:
Thanks for this feedback! I added additional order verification by the store owner in wp-admin once the order had been created.